pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -987,16 +986,7 @@ func applyOutlierDetection(cluster *apiv2.Cluster, outlier *networking.OutlierDe
 	}
 }
 
-func applyLoadBalancer(cluster *apiv2.Cluster, lb *networking.LoadBalancerSettings, port *model.Port, proxy *model.Proxy) {
-	if cluster.OutlierDetection != nil {
```

Comment:
This changed the intended behavior: if no `LoadBalancerSettings` specified, the locality setting of mesh will not be applied.

Reply:
So maybe you should set mesh locality settings to lb in `SelectTrafficPolicyComponents`

Reply:
Updated.